### PR TITLE
Use hash of entries for merge key.

### LINKED ISSUE #102: Use a hash for reference/alternate_bases when making the merge key
We currently just concatenate reference/alternate_bases together as the merge key (see [here](https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py#L120)).

This may break Dataflow if the length of reference/alternate_bases are large (larger than 1MB limit for the key). We should create a hash of these fields and use that instead. This ensures that our keys do not get arbitrarily large.

As a result, the key would be `reference_name:start_position:end_position:<hash of reference_name:alternate_bases>`. Given that we include reference/start/end as part of the key (excluding the hash) then the risk of collision is extremely low. However, we can still add an assert [here](https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py#L99) just to be sure.

### PR DESCRIPTION
**Pull request by nmousavi**
Sha256 of reference_bases and alternate_bases is used for the merge key.

fixes #102

### CODE REVIEW

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py [hunk 1]
```diff
@@ -118,13 +130,14 @@ def get_merged_variants(self, variants, unused_key=None):
...
-    yield ':'.join(
+    key = ':'.join(
         [str(x) for x in [
             variant.reference_name or '',
             variant.start or '',
             variant.end or '',
-            variant.reference_bases or '',
-            ','.join(variant.alternate_bases or [])]])
+            hashlib.sha256(variant.reference_bases or '').hexdigest(),
+            hashlib.sha256(','.join(variant.alternate_bases or [])).hexdigest()]])
```
Comment (arostamianfar) [inline gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py]: Consider making a private method _get_hash_key and use it here.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py [hunk 2]
```diff
@@ -118,13 +130,14 @@ def get_merged_variants(self, variants, unused_key=None):
...
   def get_merge_keys(self, variant):
-    yield ':'.join(
+    key = ':'.join(
         [str(x) for x in [
             variant.reference_name or '',
             variant.start or '',
             variant.end or '',
-            variant.reference_bases or '',
-            ','.join(variant.alternate_bases or [])]])
+            hashlib.sha256(variant.reference_bases or '').hexdigest(),
```
Comment (arostamianfar) [inline gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py]: I wonder if md5 is good enough here. 256 seems like an overkill :p

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py [hunk 3]
```diff
@@ -225,20 +226,28 @@ def test_get_merged_variants_move_everything_to_calls(self):
   def test_get_merge_keys(self):
     strategy = move_to_calls_strategy.MoveToCallsStrategy(None, None, None)
 
+    empty_string_hash = hashlib.sha256('').hexdigest()
```
Comment (arostamianfar) [inline gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py]: Consider making a helper method here as well. I think it's ok to use the one in the private lib too.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py (hunk 2 above)
Comment (nmousavi) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py]: Used  md5 instead.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py (hunk 1 above)
Comment (nmousavi) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py]: Done!

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py (hunk 3 above)
Comment (nmousavi) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py]: Done!

**Comment by nmousavi** [comment]
Job finished successfully on platinum dataset. 

Thanks for the review!

**Pull request merged**